### PR TITLE
e2e:chore - Create test for custom rules path flag

### PR DESCRIPTION
**Pull request by oliveirafelipezup**
Signed-off-by: oliveira.felipe <oliveira.felipe@zup.com.br>

**- What I did**
- Create test for custom rules path flag
- 
**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: e2e/commands/start/start_test.go [hunk 1]
```diff
@@ -311,4 +312,20 @@ var _ = Describe("running binary Horusec with start parameter", func() {
...
+
+		BeforeEach(func() {
+			flags = map[string]string{
+				testutil.StartFlagProjectPath:     projectPath,
+				testutil.StartFlagCustomRulesPath: customRulesJson,
+			}
+		})
+
+		It("Checks if the custom rules path property was set", func() {
+			Expect(session.Out.Contents()).To(ContainSubstring(fmt.Sprintf(`\"custom_rules_path\": \"%s\"`, customRulesJson)))
```
Comment (lucasbrunozup) [inline e2e/commands/start/start_test.go:327]: Use the `testutil.NormalizePathToAssertInJSON()` function to passed test on Windows.

`Expect(session.Out.Contents()).To(ContainSubstring(fmt.Sprintf('\"custom_rules_path\": \"%s\"', testutil.NormalizePathToAssertInJSON(customRulesJson))))`

**Review by lucasbrunozup — CHANGES_REQUESTED** [review]

File: e2e/commands/start/start_test.go (hunk 1 above)
Comment (oliveirafelipezup) [reply e2e/commands/start/start_test.go]: adjusted

**Review by lucasbrunozup — APPROVED** [review]

**Review by nathanmartinszup — APPROVED** [review]

**Comment by matheusalcantarazup** [comment]
Please, resolve the conflicts @oliveirafelipezup

**Comment by oliveirafelipezup** [comment]
> Please, resolve the conflicts @oliveirafelipezup

resolved @matheusalcantarazup

File: e2e/commands/start/start_test.go [hunk 2]
```diff
@@ -380,4 +379,19 @@ var _ = Describe("running binary Horusec with start parameter", func() {
 			Expect(session.Out.Contents()).To(ContainSubstring(fmt.Sprintf("The file or folder was ignored to send analysis:[%s]", testutil.NormalizePathToAssert(fileIgnored))))
 		})
 	})
+
+	When("--custom-path-rules is passed", func() {
+		customRulesJson := filepath.Join(os.TempDir(), fmt.Sprintf("horusec-custom-rules.json"))
```
Comment (lucasbrunozup) [inline e2e/commands/start/start_test.go:384]: The `fmt.Sprintf` is unecessary.

**Review by lucasbrunozup — CHANGES_REQUESTED** [review]

**Comment by wiliansilvazup** [comment]
Please, resolve the conflicts @oliveirafelipezup  :)

**Comment by oliveirafelipezup** [comment]
> Please, resolve the conflicts @oliveirafelipezup :)

done

**Review by wiliansilvazup — APPROVED** [review]

File: e2e/commands/start/start_test.go (hunk 2 above)
Comment (matheusalcantarazup) [reply e2e/commands/start/start_test.go:469]: This file seems not exists. I think that would be better we create a custom rule file on temp dir and use it.

**Review by matheusalcantarazup — CHANGES_REQUESTED** [review]

File: e2e/commands/start/start_test.go [hunk 3]
```diff
@@ -463,4 +463,60 @@ var _ = Describe("running binary Horusec with start parameter", func() {
...
+			Eventually(session.Wait(5 * time.Minute.Minutes()).Out).Should(gbytes.Say(`Type: Vulnerability`))
+
+		})
+	})
 })
+
+func writeJsonFile(path string) {
+	file, err := os.OpenFile(path, os.O_RDWR, os.ModePerm)
+	if err != nil {
+		Fail(fmt.Sprintf("error: %v", err))
```
Comment (matheusalcantarazup) [inline e2e/commands/start/start_test.go:495]: Change to `error to open file: %v` to more easily further debug

File: e2e/commands/start/start_test.go [hunk 4]
```diff
@@ -463,4 +463,60 @@ var _ = Describe("running binary Horusec with start parameter", func() {
...
+		},
+	}
+
+	b, err := json.Marshal(customRules)
+	if err != nil {
+		Fail(fmt.Sprintf("error: %v", err))
+	}
+
+	if _, err := file.Write(b); err != nil {
+		Fail(fmt.Sprintf("error: %v", err))
```
Comment (matheusalcantarazup) [inline e2e/commands/start/start_test.go:520]: Change to `error to write file file: %v` to more easily further debug.

File: e2e/commands/start/start_test.go [hunk 5]
```diff
@@ -463,4 +463,60 @@ var _ = Describe("running binary Horusec with start parameter", func() {
...
+			"severity":    "LOW",
+			"confidence":  "LOW",
+			"type":        "Regular",
+			"expressions": []string{".*"},
+		},
+	}
+
+	b, err := json.Marshal(customRules)
+	if err != nil {
+		Fail(fmt.Sprintf("error: %v", err))
```
Comment (matheusalcantarazup) [inline e2e/commands/start/start_test.go:516]: Change to `error to marshal custom rules: %v` to more easily further debug.

File: e2e/commands/start/start_test.go [hunk 6]
```diff
@@ -463,4 +463,60 @@ var _ = Describe("running binary Horusec with start parameter", func() {
...
+		})
+
+		It("Checks if the custom rules path property was set", func() {
+			Expect(session.Out.Contents()).To(ContainSubstring(`\"custom_rules_path\": \"%s\"`, testutil.NormalizePathToAssertInJSON(customRulesJson)))
+			Eventually(session.Wait(5 * time.Minute.Minutes()).Out).Should(gbytes.Say(`Language: Java`))
+			Eventually(session.Wait(5 * time.Minute.Minutes()).Out).Should(gbytes.Say(`Severity: LOW`))
+			Eventually(session.Wait(5 * time.Minute.Minutes()).Out).Should(gbytes.Say(`Confidence: LOW`))
+			Eventually(session.Wait(5 * time.Minute.Minutes()).Out).Should(gbytes.Say(`RuleID: HS-JAVA-99999999999`))
+			Eventually(session.Wait(5 * time.Minute.Minutes()).Out).Should(gbytes.Say(`Details: Teste QA`))
+			Eventually(session.Wait(5 * time.Minute.Minutes()).Out).Should(gbytes.Say(`Type: Vulnerability`))
```
Comment (matheusalcantarazup) [inline e2e/commands/start/start_test.go:486]: Create a variable with the timout and use the reference:
```go
timeout := 5 * time.Minute
Eventually(session.Wait(timeout).Out...
```
And btw you don't need to call `Minute.Minutes()` method, just use `N * time.Minute`.

**Review by matheusalcantarazup — CHANGES_REQUESTED** [review]

File: e2e/commands/start/start_test.go [hunk 7]
```diff
@@ -463,4 +463,61 @@ var _ = Describe("running binary Horusec with start parameter", func() {
...
+			"severity":    "LOW",
+			"confidence":  "LOW",
+			"type":        "Regular",
+			"expressions": []string{".*"},
+		},
+	}
+
+	b, err := json.Marshal(customRules)
+	if err != nil {
+		Fail(fmt.Sprintf("The following occurred in the json return: %v", err))
```
Comment (matheusalcantarazup) [inline e2e/commands/start/start_test.go:517]: "The following error occurred to marshal json: %v"

**Review by matheusalcantarazup — CHANGES_REQUESTED** [review]

**Review by matheusalcantarazup — APPROVED** [review]

**Review by lucasbrunozup — APPROVED** [review]

**Pull request merged**